tt_metal/hw/inc/blackhole/dev_mem_map.h

```diff
@@ -53,45 +51,67 @@
+
 /////////////
 // Initialization relocation L1 memory
 // Host downloads to these addresses, fw copies to destination
 // Note: using xmov to copy ncrisc to addresses above 1M hangs the chip
-#define MEM_BRISC_INIT_LOCAL_L1_BASE (MEM_TRISC2_FIRMWARE_BASE + MEM_TRISC2_FIRMWARE_SIZE)
-#define MEM_NCRISC_INIT_LOCAL_L1_BASE (MEM_BRISC_INIT_LOCAL_L1_BASE + MEM_BRISC_LOCAL_SIZE)
-#define MEM_TRISC0_INIT_LOCAL_L1_BASE (MEM_NCRISC_INIT_LOCAL_L1_BASE + MEM_NCRISC_LOCAL_SIZE)
-#define MEM_TRISC1_INIT_LOCAL_L1_BASE (MEM_TRISC0_INIT_LOCAL_L1_BASE + MEM_TRISC_LOCAL_SIZE)
-#define MEM_TRISC2_INIT_LOCAL_L1_BASE (MEM_TRISC1_INIT_LOCAL_L1_BASE + MEM_TRISC_LOCAL_SIZE)
-
-#define MEM_IERISC_INIT_LOCAL_L1_BASE (MEM_IERISC_FIRMWARE_BASE + MEM_IERISC_FIRMWARE_SIZE)
-
-#define MEM_MAP_END (MEM_TRISC2_INIT_LOCAL_L1_BASE + MEM_TRISC_LOCAL_SIZE)
+#define MEM_BRISC_INIT_LOCAL_L1_BASE_SCRATCH MEM_MAP_END
+#define MEM_NCRISC_INIT_LOCAL_L1_BASE_SCRATCH (MEM_BRISC_INIT_LOCAL_L1_BASE_SCRATCH + MEM_BRISC_LOCAL_SIZE)
+#define MEM_TRISC0_INIT_LOCAL_L1_BASE_SCRATCH (MEM_NCRISC_INIT_LOCAL_L1_BASE_SCRATCH + MEM_NCRISC_LOCAL_SIZE)
+#define MEM_TRISC1_INIT_LOCAL_L1_BASE_SCRATCH (MEM_TRISC0_INIT_LOCAL_L1_BASE_SCRATCH + MEM_TRISC_LOCAL_SIZE)
+#define MEM_TRISC2_INIT_LOCAL_L1_BASE_SCRATCH (MEM_TRISC1_INIT_LOCAL_L1_BASE_SCRATCH + MEM_TRISC_LOCAL_SIZE)
 
 /////////////
 // Stack info
 // Increasing the stack size comes at the expense of less local memory for globals
 #define MEM_BRISC_STACK_SIZE 768
 #define MEM_NCRISC_STACK_SIZE 1040
-#define MEM_IERISC_STACK_SIZE 768
 #define MEM_TRISC0_STACK_SIZE 320
 #define MEM_TRISC1_STACK_SIZE 256
 #define MEM_TRISC2_STACK_SIZE 768
 
 #define MEM_BRISC_STACK_BASE (MEM_LOCAL_BASE + MEM_BRISC_LOCAL_SIZE - MEM_BRISC_STACK_SIZE)
 #define MEM_NCRISC_STACK_BASE (MEM_LOCAL_BASE + MEM_NCRISC_LOCAL_SIZE - MEM_NCRISC_STACK_SIZE)
-#define MEM_IERISC_STACK_BASE (MEM_LOCAL_BASE + MEM_IERISC_LOCAL_SIZE - MEM_IERISC_STACK_SIZE)
 #define MEM_TRISC0_STACK_BASE (MEM_LOCAL_BASE + MEM_TRISC_LOCAL_SIZE - MEM_TRISC0_STACK_SIZE)
 #define MEM_TRISC1_STACK_BASE (MEM_LOCAL_BASE + MEM_TRISC_LOCAL_SIZE - MEM_TRISC1_STACK_SIZE)
 #define MEM_TRISC2_STACK_BASE (MEM_LOCAL_BASE + MEM_TRISC_LOCAL_SIZE - MEM_TRISC2_STACK_SIZE)
+
+/////////////
+// Padding/alignment restrictions needed in linker scripts
+// Note we pad then align so, for example, erisc can keep space between fw/kernels
+#define MEM_TENSIX_KERNEL_PAD 12
+
+
+/////////////
+// IERISC memory map
+#define MEM_IERISC_LOCAL_SIZE (8 * 1024)
+#define MEM_IERISC_FIRMWARE_SIZE (24 * 1024)
+// TODO: move this to lower memory but as w/ the TODO below, this causes problems now
+// for unknown reasons
+#define MEM_IERISC_MAILBOX_BASE 1024
+#define MEM_IERISC_MAILBOX_SIZE 3072
+#define MEM_IERISC_MAILBOX_END (MEM_IERISC_MAILBOX_BASE + MEM_IERISC_MAILBOX_SIZE)
+// TODO: this should be tightly packed against MAILBOX_END but that causes
+// problems w/ back to back runs, leaving the padding for now
```

Comment:
Regarding TODO comment @ line 104, 109.
That is because there is data in 4K block @ 0x1000 that syseng tools read back when creating ethernet map.
idle erisc code has to start at 0x2000

Reply:
Oops, is that for Blackhole? Then I am not sure. Maybe syseng has the same assumptions/requirements for BH.
What I said above is for WH.